src/invoices/InvoicesList/InvoiceLine/InvoiceLineDetailsLayer.js

```diff
@@ -1,103 +1,148 @@
 
 import {
   invoiceLineResource,
   invoiceResource,
 } from '../../../common/resources';
+import { PO_LINES_API } from '../../../common/constants';
 import InvoiceLineDetails from '../../InvoiceLineDetails';
 
-class InvoiceLineDetailsLayer extends Component {
-  static manifest = Object.freeze({
-    invoiceLine: invoiceLineResource,
-    invoice: invoiceResource,
-    query: {},
-  });
-
-  static propTypes = {
-    match: ReactRouterPropTypes.match,
-    mutator: PropTypes.object.isRequired,
-    resources: PropTypes.object.isRequired,
-    showToast: PropTypes.func.isRequired,
-  }
-
-  state = {
-    isTagsPaneOpened: false,
+const InvoiceLineDetailsLayer = ({
+  history,
+  match: { params },
+  mutator,
+  resources,
+  showToast,
+}) => {
+  const [polNumber, setPolNumber] = useState('');
+  const [isTagsPaneOpened, setTagsPaneOpened] = useModalToggle();
+
+  const fetchInvoiceLineDetails = useCallback(
+    () => {
+      mutator.invoice.reset();
+      mutator.invoiceLine.reset();
+      mutator.poLine.reset();
+      mutator.invoice.GET();
+      mutator.invoiceLine.GET()
+        .then(({ poLineId }) => {
+          if (poLineId) {
+            mutator.poLine.GET({
+              path: `${PO_LINES_API}/${poLineId}`,
+            }).then(({ poLineNumber }) => setPolNumber(poLineNumber));
+          }
+        });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [params.id],
+  );
+
+  useEffect(fetchInvoiceLineDetails, [params.id]);
+
+  const updateInvoiceLineTagList = async (invoiceLine) => {
+    await mutator.invoiceLine.PUT(invoiceLine);
+    fetchInvoiceLineDetails();
   };
 
-  toggleTagsPane = () => this.setState(({ isTagsPaneOpened }) => ({ isTagsPaneOpened: !isTagsPaneOpened }));
-
-  getInvoiceLine = () => get(this.props.resources, ['invoiceLine', 'records', 0]);
-
-  getInvoice = () => get(this.props.resources, ['invoice', 'records', 0]);
-
-  closeInvoiceLine = () => {
-    const { match: { params }, mutator } = this.props;
-    const _path = `/invoice/view/${params.id}`;
-
-    mutator.query.update({ _path });
+  const closeInvoiceLine = useCallback(
+    () => {
+      const path = `/invoice/view/${params.id}`;
+
+      history.push(path);
+    },
+    [params.id, history],
+  );
+
+  const goToEditInvoiceLine = useCallback(
+    () => {
+      const path = `/invoice/view/${params.id}/line/${params.lineId}/edit`;
+
+      history.push(path);
+    },
+    [params.id, params.lineId, history],
+  );
+
+  const deleteInvoiceLine = useCallback(
+    () => {
+      mutator.invoiceLine.DELETE({ id: params.lineId })
+        .then(() => {
+          showToast('ui-invoice.invoiceLine.hasBeenDeleted');
+          closeInvoiceLine();
+        })
+        .catch(() => {
+          showToast('ui-invoice.errors.invoiceLineHasNotBeenDeleted', 'error');
+        });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [params.lineId, mutator.invoiceLine],
```

Comment:
in our _stripes_ case I don't think that any callback should depend on `mutator.*`. Basically we always expect the same object there (but in fact it might be different, since we have no control over it)

Reply:
fixed